shell/platform/fuchsia/flutter/text_delegate_unittests.cc

```diff
@@ -0,0 +1,288 @@
+      {
+        "method": "TextInput.setClient",
+        "args": [
+           7,
+           {
+             "inputType": {
+               "name": "TextInputType.multiline",
+               "signed":null,
+               "decimal":null
+             },
+             "readOnly": false,
+             "obscureText": false,
+             "autocorrect":true,
+             "smartDashesType":"1",
+             "smartQuotesType":"1",
+             "enableSuggestions":true,
+             "enableInteractiveSelection":true,
+             "actionLabel":null,
+             "inputAction":"TextInputAction.newline",
+             "textCapitalization":"TextCapitalization.none",
+             "keyboardAppearance":"Brightness.dark",
+             "enableIMEPersonalizedLearning":true,
+             "enableDeltaModel":false
+          }
+       ]
+      }
+    )";
+    auto message = fake_platform_message_response->WithMessage(
+        kTextInputChannel, set_client_msg);
+    text_delegate_->HandleFlutterTextInputChannelPlatformMessage(
+        std::move(message));
+    RunLoopUntilIdle();
+    EXPECT_EQ(ime_service_.action_,
+              fuchsia::ui::input::InputMethodAction::NEWLINE);
+    EXPECT_FALSE(ime_service_.IsKeyboardShown());
+  }
+
+  {
+    // Verify that showing keyboard results in the correct platform effect.
+    const auto set_client_msg = R"(
+      {
+        "method": "TextInput.show"
+      }
+    )";
+    auto message = fake_platform_message_response->WithMessage(
+        kTextInputChannel, set_client_msg);
+    text_delegate_->HandleFlutterTextInputChannelPlatformMessage(
+        std::move(message));
+    RunLoopUntilIdle();
+    EXPECT_TRUE(ime_service_.IsKeyboardShown());
+  }
+
+  {
+    // Verify that hiding keyboard results in the correct platform effect.
+    const auto set_client_msg = R"(
+      {
+        "method": "TextInput.hide"
+      }
+    )";
+    auto message = fake_platform_message_response->WithMessage(
+        kTextInputChannel, set_client_msg);
+    text_delegate_->HandleFlutterTextInputChannelPlatformMessage(
+        std::move(message));
+    RunLoopUntilIdle();
+    EXPECT_FALSE(ime_service_.IsKeyboardShown());
+  }
+
+  {
+    // Update the editing state from the Fuchsia platform side.
+    fuchsia::ui::input::TextInputState state = {
+        .revision = 42,
+        .text = "Foo",
+        .selection = fuchsia::ui::input::TextSelection{},
+        .composing = fuchsia::ui::input::TextRange{},
+    };
+    auto input_event = std::make_unique<fuchsia::ui::input::InputEvent>();
+    ime_service_.client_->DidUpdateState(std::move(state),
+                                         std::move(input_event));
+    RunLoopUntilIdle();
+    EXPECT_EQ(
+        R"({"method":"TextInputClient.updateEditingState","args":[7,{"text":"Foo","selectionBase":0,"selectionExtent":0,"selectionAffinity":"TextAffinity.upstream","selectionIsDirectional":true,"composingBase":-1,"composingExtent":-1}]})",
```

Comment:
hopefully checking against this dictionary literal isn't too fragile?

Reply:
Shouldn't be that annoying, since this literal is built by this code. Whoever changes that, may need to change this too, but in that situation, the test change will need to happen anyways.

Reply:
Okay, thank you!

Reply:
Forgot to say - I didn't observe that RapidJSON arbitrarily reordering map entries when the map is built. That could have been a source of flakiness if it were the case.